features/extension_inheritance.feature

```diff
@@ -4,181 +4,32 @@ Feature: Profile extension overrides
-
-      class FeatureContext implements Context
-      {
-          private $loadedExtensions = array();
-
-          public function addExtension($extensionName) {
-              $this->loadedExtensions[] = $extensionName;
-          }
-
-          /** @When this scenario executes */
-          public function thisScenarioExecutes() {}
-
-          /** @Then the extension :name should be loaded */
-          public function theExtensionLoaded($name) {
-              PHPUnit\Framework\Assert::assertContains($name, $this->loadedExtensions);
-          }
-
-          /** @Then the extension :name should not be loaded */
-          public function theExtensionNotLoaded($name) {
-              PHPUnit\Framework\Assert::assertNotContains($name, $this->loadedExtensions);
-          }
-
-          /** @Then the extension :name should have :key set to :value */
-          public function theExtensionHas($name, $key, $value) {
-              $this->theExtensionLoaded($name);
-              $this->theExtensionNotLoaded($name);
-          }
-      }
-      """
+    Given I initialise the working directory from the "ExtensionInheritance" fixtures folder
+    And I provide the following options for all behat invocations:
+      | option      | value |
+      | --no-colors |       |
 
   Scenario: Default profile has extension loaded
-    Given a file named "behat.yml.dist" with:
-      """
-      default:
-        extensions:
-          custom_extension.php: ~
-
-      custom_profile:
-        extensions: ~
-      """
-    And a file named "features/extensions.feature" with:
-      """
-      Feature:
-        Scenario:
-          When this scenario executes
-          Then the extension "custom extension 1" should be loaded
-      """
-    When I run "behat -f progress"
+    When I run behat with the following additional options:
+      | option                                 | value |
+      | --config behat-default-extension.yaml  |       |
+      | features/extensions-default.feature    |       |
     Then it should pass
 
   Scenario: Custom profile disables all extensions
-    Given a file named "behat.yml.dist" with:
-      """
-      default:
-        extensions:
-          custom_extension.php: ~
-
-      custom_profile:
-        extensions: ~
-      """
-    And a file named "features/extensions.feature" with:
-      """
-      Feature:
-        Scenario:
-          When this scenario executes
-          Then the extension "custom extension 1" should not be loaded
-      """
-    When I run "behat -f progress -p custom_profile -vvv"
+    When I run behat with the following additional options:
+      | option                                 | value          |
+      | --config behat-default-extension.yaml  |                |
+      | --profile                              | custom_profile |
+      | -vvv                                   |                |
+      | features/extensions-disabled.feature   |                |
     Then it should pass
```

Comment:
I think this is another where it'd be good to add expected output here so we're confident the config / profile has been triggered as expected (also helps to describe what the feature does now that the files are all moved to the fixture)?

Reply:
Updated